src/components/header/_header.scss

```diff
@@ -211,6 +222,11 @@
             &--mobile {
                 display: block;
                 margin: 0 0 0.5rem;
+                .ons-header--neutral & {
+                    a {
+                        color: var(--ons-color-black);
+                    }
+                }
```

Comment:
This can be moved up to around line 209 to avoid duplicating selectors. I would also refer to the link with the proper class rather than the a element. Like this:
```
.ons-header-service-nav__link {
    color: var(--ons-color-black);
}
```